.github/workflows/tagPublish.yml

```diff
@@ -0,0 +1,85 @@
+name: package and publish to pypi
+
+on:
+  push:
+    tags:
+      - '*'
+
+jobs:
+  github-release:
+    name: Create Github release page
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.release.tag_name }}
+      - name: Set release version
+        run: echo "RELEASE_VERSION=${GITHUB_REF#refs/*/}" >> $GITHUB_ENV
+      - name: Create Github release page
+        # Create the release page first
+        run: gh release create ${{ env.RELEASE_VERSION }} --draft --notes "Auto created by tagPublish workflow."
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  wheel:
+    name: python${{ matrix.python-version }}-${{ matrix.platform.python-architecture }} ${{ matrix.platform.os }}
+    runs-on: ${{ matrix.platform.os }}
+    strategy:
+      matrix:
+        python-version: ['3.7', '3.8', '3.9', '3.10', '3.11']
+        platform: [
+        { os: "ubuntu-latest",  python-architecture: "x64", rust-target: "x86_64-unknown-linux-gnu" },
+        { os: "macOS-latest",   python-architecture: "x64", rust-target: "x86_64-apple-darwin" },
+        { os: "windows-latest", python-architecture: "x64", rust-target: "x86_64-pc-windows-msvc" },
+        { os: "ubuntu-20.04",  python-architecture: "x64", rust-target: "x86_64-unknown-linux-gnu" },
+        ]
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set release version
+        # Set release version env in all three os, the commented command only works in linux and mac.
+        run: python3 -c "import os; tag = os.environ['GITHUB_REF'].split('/')[-1]; f = open(os.environ['GITHUB_ENV'], 'a'); f.write('RELEASE_VERSION='+tag); f.close();"
+        # run: echo "RELEASE_VERSION=${GITHUB_REF#refs/*/}" >> $GITHUB_ENV
+      - name: Set up Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: ${{ matrix.python-version }}
+          architecture: ${{ matrix.platform.python-architecture }}
+      - name: Set up Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          target: ${{ matrix.platform.rust-target }}
+          override: true
+          components: rustfmt, clippy
+      - run: rustup set default-host ${{ matrix.platform.rust-target }}
+      - name: install maturin
+        run: pip install 'maturin>=0.14,<0.15'
+      - name: build wheel
+        if: ${{ matrix.platform.os != 'ubuntu-20.04'}}
+        id: build_wheel
+        run: python -u .github/workflows/build_wheel.py
+      - name: build wheel 1
+        if: ${{ matrix.platform.os == 'ubuntu-20.04'}}
+        id: build_wheel1
+        run: python -u .github/workflows/build_wheel.py "maturin","build","--release","--compatibility=manylinux_2_31","--interpreter"
```

Comment:
Here arguments are supplied , while above they are not. We should probably do it the same way for both code paths.

Reply:
Done